ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala

```diff
@@ -162,6 +177,26 @@ object ReadServiceStateUpdateComparison {
       case u: Update.CommandRejected => u.copy(recordTime = newRecordTime)
     }
 
+  private def dropFetchAndLookupByKeyNodes[Nid, Cid](
+      tx: VersionedTransaction[Nid, Cid]
+  ): VersionedTransaction[Nid, Cid] = {
+    val nodes = tx.nodes.filter {
+      case (_, _: Node.NodeFetch[Cid] | _: Node.NodeLookupByKey[Cid]) => false
+      case _ => true
+    }
+    val filteredNodes = nodes.map {
+      case (nid, node: Node.NodeExercises[Nid, Cid]) =>
+        val filteredNode = node.copy(children = node.children.filter(nodes.contains))
+        (nid, filteredNode)
+      case keep => keep
+    }
```

Comment:
Please make at least one of the two patern matching exhaustive, so we get warning when we add new type of node.